taskcluster/docker/linux/Dockerfile

```diff
@@ -0,0 +1,168 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# We use this specific version because our decision task also runs on this one.
+# We also use that same version in decisionlib.py
+FROM ubuntu:bionic-20180821
+
+MAINTAINER Edouard Oger "eoger@mozilla.com"
+
+RUN echo "trigger rebuild"
+
+# Add worker user
+
+RUN mkdir /builds && \
+    useradd -d /builds/worker -s /bin/bash -m worker && \
+    chown worker:worker /builds/worker && \
+    mkdir /builds/worker/artifacts && \
+    chown worker:worker /builds/worker/artifacts
+
+WORKDIR /builds/worker/
+
+# Configuration
+
+ENV ANDROID_BUILD_TOOLS "28.0.3"
+ENV ANDROID_SDK_VERSION "3859397"
+ENV ANDROID_PLATFORM_VERSION "28"
+
+# Set up the language variables to avoid problems (we run locale-gen later).
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
+
+# Do not use fancy output on taskcluster
+ENV TERM dumb
+
+ENV GRADLE_OPTS -Xmx4096m -Dorg.gradle.daemon=false
+
+# Used to detect in scripts whether we are running on taskcluster
+ENV CI_TASKCLUSTER true
+
+ENV \
+    # Some APT packages like 'tzdata' wait for user input on install by default.
+    # https://stackoverflow.com/questions/44331836/apt-get-install-tzdata-noninteractive
+    DEBIAN_FRONTEND=noninteractive
+
+# System.
+
+RUN apt-get update -qq \
+    && apt-get install -qy --no-install-recommends \
+        # To compile Android stuff.
+        openjdk-8-jdk \
+        git \
+        curl \
+        # Required by symbolstore.py.
+        file \
+        # Will set up the timezone to UTC (?).
+        tzdata \
+        # To install UTF-8 locales.
+        locales \
+        # For `cc` crates; see https://github.com/jwilm/alacritty/issues/1440.
+        # <TODO: Is this still true?>.
+        g++ \
+        # <TODO: Explain why we have this dependency>.
+        clang \
+        python3 \
+        python3-pip \
+        # taskcluster > mohawk > setuptools.
+        python3-setuptools \
+        # Required to extract the Android SDK/NDK.
+        unzip \
+        # Required by tooltool to extract tar.xz archives.
+        xz-utils \
+        # Required to build libs/.
+        make \
+        # Required to build sqlcipher.
+        tclsh \
+        # Required in libs/ by some scripts patching the source they download.
+        patch \
+        # For windows cross-compilation.
+        mingw-w64 \
+        ## NSS build dependencies
+        gyp \
+        ninja-build \
+        zlib1g-dev \
+        # <TODO: Delete p7zip once NSS windows is actually compiled instead of downloaded>.
+        p7zip-full \
+        ## End of NSS build dependencies
+        rsync \
+    && apt-get clean
+
+RUN pip3 install --upgrade pip
+RUN pip3 install \
+    'taskcluster>=4,<5' \
```

Comment:
Follow up in another PR: Use the latest taskcluster client.

Reply:
Noted, added to my todo-list for follow-ups.